ext/zpages/test/tracez_processor_benchmark.cc

```diff
@@ -0,0 +1,191 @@
+  }
+}
+
+/*
+ * Make many snapshots. This checks the scenario where the processor holds
+ * no spans but gets queried many times.
+ */
+BENCHMARK_DEFINE_F(TracezProcessor, BM_Snap)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    GetManySnapshots(processor, num_spans);
+  }
+}
+
+/*
+ * Make and end many empty spans. This checks the scenario where the processor holds
+ * many running and completed spans but never gets queried.
+ */
+BENCHMARK_DEFINE_F(TracezProcessor, BM_RunComplete)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> spans2;
+    StartManySpans(spans2, std::ref(tracer), num_spans);
+
+    std::thread start(StartManySpans, std::ref(spans), std::ref(tracer), num_spans);
+    EndAllSpans(spans2);
+
+    start.join();
+
+    EndAllSpans(spans);
+  }
+}
+
+/*
+ * Make many empty spans while spapshots grabbed. This checks the scenario where the
+ * processor holds many running spans and gets queried.
+ */
+BENCHMARK_DEFINE_F(TracezProcessor, BM_RunSnap)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    std::thread snapshots(GetManySnapshots, std::ref(processor), num_spans);
+    StartManySpans(spans, std::ref(tracer), num_spans);
+
+    snapshots.join();
+    EndAllSpans(spans);
+  }
+}
+
+/*
+ * Make many empty spans end while snapshots are being grabbed. This checks the scenario
+ * where the processor doesn't make new spans, but existing spans complete while they're
+ * queried.
+ */
+BENCHMARK_DEFINE_F(TracezProcessor, BM_SnapComplete)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    StartManySpans(spans, std::ref(tracer), num_spans);
+
+    std::thread snapshots(GetManySnapshots, std::ref(processor), num_spans);
+    EndAllSpans(spans);
+
+    snapshots.join();
+  }
+
+}
+
+/*
+ * Make many empty spans and end some, all while snapshots are being grabbed. This
+ * checks the scenario where the processor makes new spans, other spans complete,
+ * and all spans are queried. This is the case most similar to real situations.
+ */
+BENCHMARK_DEFINE_F(TracezProcessor, BM_RunSnapComplete)(benchmark::State &state)
+{
+  const int num_spans = state.range(0);
+  for (auto _ : state)
+  {
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> spans2;
+
+    StartManySpans(spans, std::ref(tracer), num_spans);
+
+    std::thread start(StartManySpans, std::ref(spans2), std::ref(tracer), num_spans);
+    std::thread snapshots(GetManySnapshots, std::ref(processor), num_spans);
+    EndAllSpans(spans);
+
+    start.join();
+    snapshots.join();
+
+    EndAllSpans(spans2);
+  }
+}
+
+/////////////////////// RUN BENCHMARKS ///////////////////////////
+
+BENCHMARK_REGISTER_F(TracezProcessor, BM_Run)->Arg(10)->Arg(1000);
```

Comment:
Nit: If for all benchmarks the argument is number of spans, you can add here a comment stating that.

Reply:
okay, thank you!!